src/main/java/org/jboss/ejb/protocol/remote/EJBServerChannel.java

```diff
@@ -713,10 +713,10 @@ public void convertToStateful(@NotNull final SessionID sessionId) throws Illegal
                 if (1 <= version && version <= 2) {
                     final Marshaller marshaller = marshallerFactory.createMarshaller(configuration);
                     marshaller.start(new NoFlushByteOutput(Marshalling.createByteOutput(os)));
-                    if (weakAffinityUpdate != null) {
-                        marshaller.writeObject(weakAffinityUpdate);
+                    if (strongAffinityUpdate != null) {
+                        marshaller.writeObject(strongAffinityUpdate);
                     } else {
-                        marshaller.writeObject(new NodeAffinity(channel.getConnection().getEndpoint().getName()));
+                        marshaller.writeObject(Affinity.NONE);
```

Comment:
Are you sure this is correct?  It should never be the case that a @Stateful EJB has no strong affinity.